quarkus/extensions/src/main/java/org/keycloak/provider/quarkus/QuarkusStartupObserver.java

```diff
@@ -34,6 +53,55 @@ private void startupEvent(@Observes StartupEvent event) {
         if (command != null) {
             command.run();
         }
+
+        createAdminUser();
+    }
+
+    private void createAdminUser() {
+        String adminUserName;
+        String adminPassword;
+
+        if ((adminUserName = System.getenv(KEYCLOAK_ADMIN_ENV_VAR)) == null
+                || (adminPassword = System.getenv(KEYCLOAK_ADMIN_PASSWORD_ENV_VAR)) == null) {
+            return;
+        }
+
+        KeycloakSessionFactory sessionFactory = (KeycloakSessionFactory) servletContext
+                .getAttribute(KeycloakSessionFactory.class.getName());
+        KeycloakSession session = sessionFactory.create();
+        KeycloakTransactionManager transaction = session.getTransactionManager();
+
+        try {
+            transaction.begin();
+
+            RealmModel realm = session.realms().getRealmByName(DEFAULT_REALM);
+            UserProvider users = session.users();
+
+            if (users.getUserByUsername(adminUserName, realm) != null) {
+                ServicesLogger.LOGGER.notCreatingExistingUser(adminUserName);
+            } else {
+                UserModel user = users.addUser(realm, adminUserName);
```

Comment:
You can use https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/services/managers/ApplianceBootstrap.java#L95 for this